app/models/long_submission.rb

```diff
@@ -40,11 +40,15 @@ class LongSubmission < ActiveRecord::Base
 
   validates :score,
             numericality: { allow_nil: true, greater_than_or_equal_to: 0 }
+  def get_score_text
+    return "-" if score.nil?
+    score.to_s
+  end
 
-  SCORE_HASH = [*0..LongProblem::MAX_MARK].each_with_object({}) do |item, memo|
-    memo[item] = item.to_s
+  def text_to_score(x) 
```

Comment:
Isn't this supposed to be a class method?

Reply:
It was, but when i realized that the long_submission score must not exceed max_score, i need the max_score so i bind it then become object method, i guess

Reply:
I see. Looking at the usages of this method I would think that checking for exceeding max_score might be misleading. Imagine if you are marking a submission and you accidentally put score 8 when the max score is 7. Now it will be stored as nil and the marker will not be alerted for this. I think it's much better to have validations for the score of the long submission to not exceed the long problem's max score instead.